### PR TITLE
Fix panic in tedge_config macro when `-` present in field name

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

`#[tedge_config(rename = "name-with-hyphens")]` panicked because of hyphens in the rename field. This occured because a test that was being generated to test that field did not remove hyphens from the identifier name, e.g:

> error: proc macro panicked
> ...
>     = help: message:
`"example_value_can_be_deserialized_for_run_name-with-hyphens_example_0"` is not a valid identifier

Was fixed by replacing hyphens with underscores in identifier name.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/tedge_config_macros/impl/src/lib.rs [hunk 1]
```diff
@@ -188,4 +188,18 @@ mod tests {
...
     }
+
+    #[test]
+    fn can_contain_hyphen_separated_fields() {
+        generate_configuration(quote! {
+            device: {
+                #[tedge_config(rename = "type")]
+                ty: String,
+
+                #[tedge_config(rename = "hyphen-separated-field", example = "hsf")]
```
Comment (Bravo555) [inline crates/common/tedge_config_macros/impl/src/lib.rs:199]: One thing I noticed is that the panic was not triggered with `#[tedge_config(rename = "hyphen-separated-field")]` and I also had to add `example = hsf` in the macro. Is this intended behaviour?

Comment (jarhodes314) [reply crates/common/tedge_config_macros/impl/src/lib.rs:199]: Yes, the generated tests exist to verify the examples are valid, so there is no test if no example is provided, and if there was a second example two tests would be generated.

**Review by jarhodes314 — APPROVED** [review]
This looks good to me. Out of interest, what are you intending to use hyphens for? I'm just wondering as we use underscores elsewhere in tedge.toml, and using a different casing style may be confusing.

**Comment by reubenmiller** [comment]
> This looks good to me. Out of interest, what are you intending to use hyphens for? I'm just wondering as we use underscores elsewhere in tedge.toml, and using a different casing style may be confusing.

Yes I would not advise to use hypens as this makes translating the configuration name to the environment variable equivalent more complicated as environment variables should only match the following regex (for compatibility): `^[A-Z][A-Z0-9]+$`

**Comment by Bravo555** [comment]
> Out of interest, what are you intending to use hyphens for? I'm just wondering as we use underscores elsewhere in tedge.toml, and using a different casing style may be confusing.

I was just testing how the config options would look like with hyphens, and stumbled on this bug. I don't intend to use them for any configuration options.

**Comment by jarhodes314** [comment]
> > This looks good to me. Out of interest, what are you intending to use hyphens for? I'm just wondering as we use underscores elsewhere in tedge.toml, and using a different casing style may be confusing.
> 
> Yes I would not advise to use hypens as this makes translating the configuration name to the environment variable equivalent more complicated as environment variables should only match the following regex (for compatibility): `^[A-Z][A-Z0-9_]*$`

Obviously this bug was never the right behaviour, but I'm wondering whether characters that can't be (widely) used in environment variable names should be explicitly rejected with an appropriate compiler error explaining why we don't support them, in case someone accidentally uses incorrect casing in the future.

**Pull request merged**